ooniprobe/Model/Database/Measurement.mm

```diff
@@ -112,61 +118,9 @@ -(void)deleteObject{
 }
 
 -(void)getExplorerUrl:(void (^)(NSString*))successcb onError:(void (^)(NSError*))errorcb {
-    NSURLComponents *components = [[NSURLComponents alloc] init];
-    components.scheme = @"https";
-    components.host = @"api.ooni.io";
-    components.path = @"/api/v1/measurements";
-    NSURLQueryItem *reportIdItem = [NSURLQueryItem
-                                    queryItemWithName:@"report_id"
-                                    value:self.report_id];
-    //web_connectivity is the only test using input for now
-    if ([self.test_name isEqualToString:@"web_connectivity"]){
-        NSURLQueryItem *urlItem = [NSURLQueryItem
-                                   queryItemWithName:@"input"
-                                   value:self.url_id.url];
-        components.queryItems = @[ reportIdItem, urlItem ];
-    }
-    else
-        components.queryItems = @[ reportIdItem ];
-
-    NSURL *url = components.URL;
-    NSURLSessionDataTask *downloadTask = [[NetworkSession getSession]
-     dataTaskWithURL:url
-     completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
-         [self getExplorerUrlCallback:data response:response error:error
-                          onSuccess:successcb onError:errorcb];
-     }];
-    [downloadTask resume];
-}
-
-- (void)getExplorerUrlCallback:(NSData *)data
-                    response:(NSURLResponse *)response
-                       error:(NSError *)error
-                   onSuccess:(void (^)(NSString*))successcb
-                     onError:(void (^)(NSError*))errorcb {
-    if (error != nil) {
-        errorcb(error);
-        return;
-    }
-    NSDictionary *dic = [NSJSONSerialization JSONObjectWithData:data options:0 error:&error];
-    if (error != nil) {
-        errorcb(error);
-        return;
-    }
-    NSArray *resultsArray = [dic objectForKey:@"results"];
-    /*
-     Checking if the array is longer than 1.
-     https://github.com/ooni/probe-ios/pull/293#discussion_r302136014
-     */
-    if ([resultsArray count] != 1 ||
-        ![[resultsArray objectAtIndex:0] objectForKey:@"measurement_url"]) {
-        errorcb([NSError errorWithDomain:@"io.ooni.api"
-                                    code:ERR_JSON_EMPTY
-                                userInfo:@{NSLocalizedDescriptionKey:@"Modal.Error.JsonEmpty"
-                                           }]);
-        return;
-    }
-    successcb([[resultsArray objectAtIndex:0] objectForKey:@"measurement_url"]);
+    [OONIApi getExplorerUrl:self.report_id
+                    withUrl:self.url_id!=nil?self.url_id.url:nil
```

Comment:
This could be simplified as: `self.url_id`

Reply:
I don't need `self.url_id` but `self.url_id.url` that is a value inside the `Url` object

Reply:
Ah, so the code was not readable, please: `((self.url_id != nil) ? self.url_id.url : nil)`

Reply:
Got it! done